pkg/providers/tinkerbell/hardware/machine.go

```diff
@@ -29,6 +35,39 @@ type Machine struct {
 	VLANID       string `csv:"vlan_id, omitempty"`
 }
 
+// NewMachineFromHardware creates a new Machine from the given hardware and it's dependencies, if any.
+func NewMachineFromHardware(hw tinkv1alpha1.Hardware, rm *rufiov1alpha1.Machine, s *corev1.Secret) Machine {
+	var bmcIPAddress, bmcUsername, bmcPassword string
+	if rm != nil && s != nil {
+		bmcIPAddress = rm.Spec.Connection.Host
+		bmcUsername = string(s.Data["username"])
+		bmcPassword = string(s.Data["password"])
```

Comment:
I think these are base64 encoded. Does the Machine struct below expect these to be in base64 format?

Reply:
The machine struct expects the unencoded data, and `Data` is a map[string][]byte so I think `s.Data["username"]` returns the expected output here. Through manual testing, I see that the output YAML is generated as expected